pkg/server/backuprestoreserver.go

```diff
@@ -153,6 +153,50 @@ func (b *BackupRestoreServer) startHTTPServer(initializer initializer.Initialize
 	return handler
 }
 
+func waitUntilEtcdRunning(ctx context.Context, etcdConnectionConfig *brtypes.EtcdConnectionConfig, logger *logrus.Logger) error {
+	ticker := time.NewTicker(4 * time.Second)
+	defer ticker.Stop()
+	logger.Info("Checking if etcd is running")
+	for !isEtcdRunning(ctx, 2*time.Second, etcdConnectionConfig, logger) {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+		}
+	}
+	logger.Info("Etcd is now running. Continuing br startup")
+	return nil
+}
+
+func isEtcdRunning(ctx context.Context, timeout time.Duration, etcdConnectionConfig *brtypes.EtcdConnectionConfig, logger *logrus.Logger) bool {
+	var endPoint string
+
+	factory := etcdutil.NewFactory(*etcdConnectionConfig)
+	client, err := factory.NewMaintenance()
+	if err != nil {
+		logger.Errorf("failed to create etcd maintenance client: %v", err)
+		return false
+	}
+	defer client.Close()
+
+	if len(etcdConnectionConfig.Endpoints) > 0 {
+		endPoint = etcdConnectionConfig.Endpoints[0]
+	} else {
+		logger.Errorf("etcd endpoints are not passed correctly")
+		return false
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	_, err = client.Status(ctx, endPoint)
+	if err != nil {
+		logger.Errorf("failed to get status of etcd endPoint: %v with error: %v", endPoint, err)
```

Comment:
```suggestion
	if len(etcdConnectionConfig.Endpoints) == 0 {
		logger.Errorf("etcd endpoints are not passed correctly")
		return false

	}
	endpoint := etcdConnectionConfig.Endpoints[0]

	ctx, cancel := context.WithTimeout(ctx, timeout)
	defer cancel()

	_, err = client.Status(ctx, endpoint)
	if err != nil {
		logger.Errorf("failed to get status of etcd endPoint: %v with error: %v", endpoint, err)
```

Can we follow consistent of checking error and return false if any error

WDYT?